.github/workflows/python-dev.yml

```diff
@@ -75,15 +75,30 @@ jobs:
         python -m pip install --upgrade pip setuptools wheel
         python -m pip install -i https://pypi.anaconda.org/scipy-wheels-nightly/simple numpy
         python -m pip install git+https://github.com/nedbat/coveragepy.git
+        python -m pip install python-dateutil pytz cython
+        # TODO: update when upstream releases fixes
+        python -m pip install "meson[ninja] @ git+https://github.com/mesonbuild/meson.git@master"
+        python -m pip install "git+https://github.com/mesonbuild/meson-python.git@main"
+        python -m pip install hypothesis==6.52.1 pytest>=6.2.5 pytest-xdist pytest-cov pytest-asyncio>=0.17
         python -m pip install versioneer[toml]
-        python -m pip install python-dateutil pytz cython hypothesis==6.52.1 pytest>=6.2.5 pytest-xdist pytest-cov pytest-asyncio>=0.17
         python -m pip list
 
-    # GH 47305: Parallel build can cause flaky ImportError from pandas/_libs/tslibs
+      # Sigh, someone (numpy?) is depending on mingw, which pandas doesn't compile with.
```

Comment:
Interesting we can't be compiled by mingw. An alternate would make that possible

Reply:
SciPy builds all Windows wheels with MinGW; NumPy builds 32-bit wheels with it. Not sure what the problem is here exactly, NumPy using MinGW shouldn't prevent Pandas from building with MSVC. 

That said, it may be useful to just fix the problem in Pandas that prevents building with MinGW? It probably indicates an actual problem in C/Cython code somewhere.

Reply:
It's worth reconsidering if we even need to transpile that one particular module to C++. Not sure why that was historically done.

Reply:
I think we use deque from STL.

Reply:
Ah nice. Python has its own deque in collections I wonder if we couldn't use that instead. 

Outside the scope of this PR but something we can look at separately